protocol-units/settlement/mcr/setup/src/local.rs

```diff
@@ -1,65 +1,207 @@
+	) -> impl Future<Output = Result<(Config, tokio::task::JoinHandle<Result<String, anyhow::Error>>), anyhow::Error>> + Send {
+		//define a temporary chain Id for Anvil
+		let mut rng = thread_rng(); // rng is not send.
+		let id: u16 = rng.gen_range(100, 32768);
+		let chain_id = id.to_string();
+		config.eth_chain_id = id as u64;
+
+		tracing::info!("Init Settlement local conf");
+
 		async move {
-			// Can't use rand 0.7 while k256 is on rand 0.6
-			let mut rng = k256::elliptic_curve::rand_core::OsRng;
 
-			info!("setting up MCR Ethereum client");
+			tracing::info!("Run Settlement local conf: {:?}", config.signer_private_key);
+			
+			//start local process and deploy smart contract.
+			//define working directory of Anvil
+			let mut path = dot_movement.get_path().to_path_buf();
+			path.push("anvil/mcr");
+			path.push(chain_id.clone());
+			tokio::fs::create_dir_all(&path).await?;
+			path.push("anvil.json");
 
-			if config.rpc_url.is_none() {
-				config.rpc_url = Some(format!("http://localhost:{}", self.eth_rpc_port));
-			}
-			if config.ws_url.is_none() {
-				config.ws_url = Some(format!("http://localhost:{}", self.eth_ws_port));
-			}
-			if config.signer_private_key.is_none() {
-				let key = SigningKey::random(&mut rng);
-				let key_bytes = key.to_bytes();
-				let private_key = hex::encode(key_bytes.as_slice());
-				config.signer_private_key = Some(private_key);
+			let anvil_path = path.to_string_lossy().to_string();
+
+			let (anvil_cmd_id, anvil_join_handle) = spawn_command(
+				"anvil".to_string(),
+				vec![
+					"--chain-id".to_string(),
+					config.eth_chain_id.to_string(),
+					"--config-out".to_string(),
+					anvil_path.clone(),
+					"--port".to_string(),
+					config.eth_rpc_connection_port.to_string(),
+					"--steps-tracing".to_string()
+				],
+			)
+			.await?;
+			//wait Anvil to start
+			let mut counter = 0;
+			loop {
+				if counter > 10 {
+					return Err(anyhow!("Anvil didn't start in time"));
+				}
+				counter += 1;
+				if path.exists() {
+					break;
+				}
+				let _ = tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
 			}
 
-			Ok(config)
+			// Deploy MCR smart contract.
+			let anvil_addresses =
+				mcr_settlement_client::eth_client::read_anvil_json_file_addresses(
+					&*anvil_path,
+				)?;
+			config.governor_private_key = anvil_addresses.get(0).ok_or(
+				anyhow!("Governor private key not found in Anvil addresses"),
+			)?.private_key.clone();
+			let governor_address = anvil_addresses.get(0).ok_or(
+				anyhow!("Governor address not found in Anvil addresses"),
+			)?.address.clone();
+
+			// todo: make sure this shows up in the docker container as well
+			let mut solidity_path = std::env::current_dir()?;
+			solidity_path.push("protocol-units/settlement/mcr/contracts");
+
+			let solidity_path = solidity_path.to_string_lossy();
+			tracing::info!("solidity_path: {:?}", solidity_path);
+			let output_exec = run_command(
```

Comment:
Add it above here.

Reply:
`anvil build`

Reply:
what did you mean here?

Reply:
See how there's a `run_command` for the `forge sciprt` script. Add a `forge build/compile` above it.